arrows/vxl/color_commonality_filter.cxx

```diff
@@ -0,0 +1,600 @@
+
+  config->set_value(
+    "color_resolution_per_channel",
+    d->color_resolution_per_chan,
+    "Resolution of the utilized histogram (per channel) if the input "
+    "contains 3 channels. Must be a power of two." );
+  config->set_value(
+    "intensity_resolution", d->intensity_resolution,
+    "Resolution of the utilized histogram if the input "
+    "contains 1 channel. Must be a power of two." );
+  config->set_value(
+    "output_scale", d->settings.output_scale_factor,
+    "Scale the output image (typically, values start in the range [0,1]) "
+    "by this amount. Enter 0 for type-specific default." );
+  config->set_value(
+    "grid_image", d->settings.grid_image,
+    "Instead of calculating which colors are more common "
+    "in the entire image, should we do it for smaller evenly "
+    "spaced regions?" );
+  config->set_value(
+    "grid_resolution_height",
+    d->settings.grid_resolution_height,
+    "Divide the height of the image into x regions, if enabled." );
+  config->set_value(
+    "grid_resolution_width",
+    d->settings.grid_resolution_width,
+    "Divide the width of the image into x regions, if enabled." );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+color_commonality_filter
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Start with our generated vital::config_block to ensure that assumed values
+  // are present. An alternative would be to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for ccf
+  d->color_resolution_per_chan =
+    config->get_value< unsigned >(
+      "color_resolution_per_channel", d->color_resolution_per_chan );
+  d->intensity_resolution =
+    config->get_value< unsigned >(
+      "intensity_resolution", d->intensity_resolution );
+  d->settings.output_scale_factor =
+    config->get_value< unsigned >(
+      "output_scale", d->settings.output_scale_factor );
+  d->settings.grid_image =
+    config->get_value< bool >(
+      "grid_image", d->settings.grid_image );
+  d->settings.grid_resolution_width =
+    config->get_value< unsigned >(
+      "grid_resolution_width", d->settings.grid_resolution_width );
+  d->settings.grid_resolution_height =
+    config->get_value< unsigned >(
+      "grid_resolution_height", d->settings.grid_resolution_height );
+
+  d->color_resolution = d->color_resolution_per_chan *
+                        d->color_resolution_per_chan *
+                        d->color_resolution_per_chan;
+
+  d->color_histogram.resize( d->color_resolution, 0 );
+  d->intensity_histogram.resize( d->intensity_resolution, 0 );
+}
+
+// ----------------------------------------------------------------------------
+bool
+color_commonality_filter
+::check_configuration( vital::config_block_sptr config ) const
```

Comment:
Should this maybe do the same 'merge atop the default config' that `set_configuration` does?

Reply:
I see the argument for that. However, I don't think I've seen this in any other filters so maybe it's worth just doing a batch PR to fix a number of them at once. 

In practice, my guess is check config is called after merging with the default, because I've never had key errors.

Reply:
Actually, I think "in practice" it is only unit tests that actually call anyone's `set_configuration` :slightly_smiling_face:. And they probably don't check empty/default configurations.

I could see punting to a PR that fixes a bunch of these...